src/hooks/use-query.ts

```diff
@@ -90,6 +147,60 @@ export function useGroupProposals(groupId?: string) {
   })
 }
 
+export function useGroupHistoricalProposals(groupId?: string) {
+  const { data: policies, isLoading } = useGroupPolicies(groupId)
+  const policyIds = policies?.map((p) => p.address) || []
+  const { client } = useGraphQLClient()
+  return useQuery({
+    queryKey: ['historicalProposals', groupId],
+    queryFn: async () => {
+      if (!client) {
+        throw Error('graphql client not initialized')
+      }
+      const result = []
+      for (const address of policyIds) {
+        const res = await client.request(ProposalsByGroupPolicyAddressDocument, {
+          groupPolicyAddress: address,
+        })
+        if (!!res.allProposals && !!res.allProposals.nodes) {
+          const { nodes } = res.allProposals
+          for (const node of nodes) {
+            const proposal = getFragmentData(ProposalItemFragmentDoc, node)
+            if (proposal) {
+              const executorResult =
+                {
+                  PROPOSAL_EXECUTOR_RESULT_UNSPECIFIED: 0,
+                  PROPOSAL_EXECUTOR_RESULT_NOT_RUN: 1,
+                  PROPOSAL_EXECUTOR_RESULT_SUCCESS: 2,
+                  PROPOSAL_EXECUTOR_RESULT_FAILURE: 3,
+                  UNRECOGNIZED: -1,
+                }[proposal.executorResult] || -1
+              const uiProposal: UIProposal = {
+                metadata: JSON.parse(proposal.metadata) as UIProposalMetadata,
+                executorResult,
+                id: proposal.id,
+                groupPolicyAddress: proposal.groupPolicyAddress,
+                proposers: proposal.proposers as string[],
+                groupVersion: proposal.groupVersion,
+                groupPolicyVersion: proposal.groupPolicyVersion,
+                status: ProposalStatus[proposal.status as keyof typeof ProposalStatus],
+                finalTallyResult: proposal.finalTallyResult,
+                messages: proposal.messages,
+                submitTime: proposal.submitTime,
+                votingPeriodEnd: proposal.votingPeriodEnd,
+                historical: true,
+              }
```

Comment:
good catch, i'll do that, and yes i think we should convert it the same way

Reply:
591b7cf

Reply:
i decided to put this in a new file called:
```src/graphql/historicalProposal.utils.ts```

my thinking is that it is nice to have this there since it's related to converting results of graphql queries.
that said, maybe it's better have it elsewhere
spoons go here, forks go there, whatever 😉

the other place i was considering was in `src/api` but i held off on that.
it looks like at present this is all stuff related to ledger queries.
arguably the home for this function could be here:
```src/api/proposal.utils.ts```

but my thinking is that some lines drawn between new utilities for graphql queries vs ledger stuff is helpful.

open to other ideas!